STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java

```diff
@@ -400,8 +401,12 @@ public Object invoke (Object proxy, java.lang.reflect.Method method, Object[] ar
                             }
                         }
                     }
-    
-                    return method.invoke(_theObject, args);
+
+                    try {
+                        return method.invoke(_theObject, args);
+                    } catch (InvocationTargetException e) {
+                        throw e.getCause() != null ? e.getCause() : e;
```

Comment:
No, this is application code throwing an exception. It is up to the developer to catch his own exception and log accordingly.